File: mockez-back/src/main/java/com/mockez/domain/model/entity/Group.java
package com.mockez.domain.model.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import java.util.List;
import java.util.UUID;

import static java.util.Collections.emptyList;

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "[GROUP]")
@ToString(callSuper = true)
@SuperBuilder(toBuilder = true)
public class Group extends Base {

    @Id
    @Column(name = "ID", updatable = false)
    private UUID id;

    @Column(name = "NAME", nullable = false)
    private String name;

    @Column(name = "DESCRIPTION", nullable = false, length = 1000)
    private String description;

    @ToString.Exclude
    @OneToMany(mappedBy = "group")
    private List<Project> projects = emptyList();

    @ToString.Exclude
    @OneToMany(mappedBy = "group")
    private List<GroupMember> groupMembers = emptyList();

}

File: mockez-front/src/app/core/model/group.ts
import { Base } from '@core/model/base';
import { Project } from '@core/model/project';
import { GroupMember } from '@core/model/group-member';

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

export class Group extends Base {

  public id: string | undefined = undefined;
  public name: string | undefined = undefined;
  public description: string | undefined = undefined;
  public projects: Project[] = [];
  public groupMembers: GroupMember[] = [];

}
